src/webgpu/api/operation/texture_view/write.spec.ts

```diff
@@ -13,20 +16,354 @@ TODO: Write helper for this if not already available (see resource_init, buffer_
+        const multisampleCount = method === 'render-pass-store' ? sampleCount : 4;
+
+        const pipeline = t.device.createRenderPipeline({
+          layout: 'auto',
+          vertex: {
+            module: t.device.createShaderModule({
+              code: kFullscreenQuadVertexShaderCode,
+            }),
+          },
+          fragment: {
+            module: t.device.createShaderModule({
+              code: `
+                @fragment fn main(
+                  @builtin(position) fragCoord: vec4<f32>,
+                ) -> @location(0) ${vecType} {
+                  const src = ${kColorArrayShaderString};
+                  let coord = vec2u(fragCoord.xy);
+                  let idx = coord.x + coord.y * ${kTextureSize};
+                  return src[idx % ${kColors.length}];
+                }`,
+            }),
+            targets: [
+              {
+                format,
+              },
+            ],
+          },
+          multisample: {
+            count: multisampleCount,
+          },
+        });
+        const commandEncoder = t.device.createCommandEncoder();
+        const pass = commandEncoder.beginRenderPass({
+          colorAttachments: [
+            {
+              view: targetView,
+              resolveTarget: resolveView,
+              loadOp: 'clear',
+              storeOp: 'store',
+            },
+          ],
+        });
+        pass.setPipeline(pipeline);
+        pass.draw(6);
+        pass.end();
+        t.device.queue.submit([commandEncoder.finish()]);
+      }
+      break;
+    default:
+      unreachable();
+  }
+
+  return expectedTexelView;
+}
 
 g.test('format')
   .desc(
     `Views of every allowed format.
 
+Read values from color array in the shader, and write it to the texture view via different write methods.
+
 - x= every texture format
 - x= sampleCount {1, 4} if valid
 - x= every possible view write method (see above)
 `
   )
-  .unimplemented();
+  .params(u =>
+    u //
+      .combine('method', kTextureViewWriteMethods)
+      .combine('format', kRegularTextureFormats)
+      .combine('sampleCount', [1, 4])
+      .filter(({ format, method, sampleCount }) => {
+        const info = kTextureFormatInfo[format];
+
+        if (sampleCount > 1 && !info.multisample) {
+          return false;
+        }
+
+        switch (method) {
+          case 'storage-write-compute':
+          case 'storage-write-fragment':
+            return info.color?.storage && sampleCount === 1;
+          case 'render-pass-store':
+            return !!info.colorRender;
+          case 'render-pass-resolve':
+            return !!info.colorRender?.resolve && sampleCount === 1;
+        }
+        return true;
+      })
+  )
+  .beforeAllSubcases(t => {
+    const { format, method, sampleCount } = t.params;
+    t.skipIfTextureFormatNotSupported(format);
+
+    // MAINTENANCE_TODO: Remove this skipping when TexelRepresentation.numericRange is made per-component.
```

Comment:
The MAINTENANCE_TODOs should be actual TODOs (in the test description) because they impact test coverage, not just code refactoring.